agent/agent.go

```diff
@@ -267,14 +289,27 @@ func (r *Agent) setupSink(ctx context.Context, sr recipe.PluginRecipe, stream *s
 		)
 	}
 	stream.subscribe(func(records []models.Record) error {
+		pluginInfo.BatchSize = len(records)
+
 		err := r.retrier.retry(
 			ctx,
-			func() error { return sink.Sink(ctx, records) },
+			func() error {
+				if r.otelEnabled {
+					sink, err = otelmw.WithSinkMW(sink, sr.Name)
+					if err != nil {
+						return err
+					}
+				}
```

Comment:
We should do this after line 271 and not within each retry attempt.

Reply:
updated